### DISCUSSION TITLE
Handling Element Visibility Based on Page Width in Next.js to Prevent Page Flickering

### DISCUSSION THREAD

**Discussion by a145789** [vercel/next.js#53236] [Help] [answered]
### Summary

As a newcomer to Next.js, I encountered a challenge in my project where I needed to determine whether to display a particular element based on the current page width. However, since the page width is not available on the server-side and defaults to a predefined value, the page flickers when the actual width is obtained on the client-side. I am sincerely seeking assistance from the community and would appreciate any effective solutions to this issue. Thank you!

**Comment by icyJoseph** [comment, marked as answer]
Try to solve this using CSS first. 

> I needed to determine whether to display a particular element based on the current page width.

Exactly that can be done with a media query and the display property. You could also use CSS with the visibility property, but that'll take up "space" in the layout, but won't show the contents. This way the HTML related to this component can also be (pre-)rendered server side.

If you need to further fetch data/do login, or would like to prevent it, based on visibility, you can use JavaScript inside a use-effect, to trigger work depending on whether or no the element is visible.
